### PR TITLE
feat: Support for AMT_AlarmClockService and IPS_AlarmClockOccurrence

### PR DESCRIPTION
**Pull request by orinem**
## PR Checklist

- [ X] Unit Tests have been added for new changes

## Note

This is the first step in adding support for the AMT Alarm Clock Service in OpenAMT.  I'm submitting this now to avoid potential duplication of effort.  The next step will be to add APIs to MPS that will Enumerate alarms, add an alarm or delete an alarm.

## What are you changing?

Added support for Get and AddAlarm for AMT_AlarmClockService.
Added support for Enumerate, Pull and Delete for IPS_AlarmClockOccurrence.

## Anything the reviewer should know when reviewing this PR?

Release isn't supported on IPS_AlarmClockOccurrence as there are a maximum of 5 alarms. The first Pull should request at least 5 elements which will mean the EnumerationContext will automatically become invalid after the request.  Alternatively, the user could call Pull until EndOfSequence is returned.

The XML returned was based on that generated by the AMT SDK C# library sample and has been tested to successfully add an alarm using Postman.

The IPS tests were using hard-coded MessageIds which failed when adding a new test.  Rather than changing the hard-coded values, I changed the tests to use the same method of generating correct responses as the CIM tests, i.e. using a MessageId variable that is incremented by each test.

The AddAlarm test builds up the correct response line by line to make it more readable.  It would be a very long line otherwise.

Note that the Alarm Clock Service doesn't understand seconds and rejects any request with seconds in the time other than '00'.  Similarly, the Interval for periodic alarms is in minutes.

The changes to package-lock.json are to do with eslint and can be ignored.

### CODE REVIEW

**Comment by orinem** [comment]
The workflow failures: 

```
Command failed with exit code 128: git log --format=%H--------->hash---------%B%n--------->commit--------- 1552a879bfd684675fec866c8c2da3f3ed422455^1..886cf3df09cbcc137fd443825aaa9310730a277d --first-parent
[13](https://github.com/open-amt-cloud-toolkit/wsman-messages/runs/7477536995?check_suite_focus=true#step:5:14)
fatal: ambiguous argument '1552a879bfd684675fec866c8c2da3f3ed422455^1..886cf3df09cbcc137fd443825aaa9310730a277d': unknown revision or path not in the working tree.
[14](https://github.com/open-amt-cloud-toolkit/wsman-messages/runs/7477536995?check_suite_focus=true#step:5:15)
```


Would to the best of my research, seem to do with:

```
         fetch-depth: 0
```

in semantic.yml causing the clone not to fetch the 'main' branch and are therefore false positives.  It all passes commitlint locally.

File: src/models/ips_models.ts [hunk 1]
```diff
@@ -35,3 +35,11 @@ export interface CancelOptIn_OUTPUT {
 export interface SendOptInCode_OUTPUT {
   SendOptInCode_OUTPUT: ReturnValue
 }
+
+export interface IPS_AlarmClockOccurrence extends CIM_ManagedElement {
+  ElementName?: string
```
Comment (rsdmike) [inline src/models/ips_models.ts:40]: Shouldn't need ElementName here as it is in CIM_ManagedElement

File: src/ips/models.ts [hunk 2]
```diff
@@ -33,3 +33,11 @@ export interface CancelOptIn_OUTPUT {
 export interface SendOptInCode_OUTPUT {
   SendOptInCode_OUTPUT: ReturnValue
 }
+
+export interface AlarmClockOccurrence extends ManagedElement {
+  ElementName?: string
```
Comment (rsdmike) [inline src/ips/models.ts:38]: Shouldn't need ElementName here as it is in ManagedElement

File: src/ips/messages.test.ts [hunk 3]
```diff
@@ -5,22 +5,61 @@
...
+    it('should throw error if selector is missing from ips_AlarmClockOccurrence Delete method', () => {
+      expect(() => { ipsClass.AlarmClockOccurrence(Methods.DELETE) }).toThrow(WSManErrors.SELECTOR)
+    })
+    it('should throw error if an unsupported method is called', () => {
+      /* Not supporting GET or PUT */
+      expect(() => { castedIPSClass.AlarmClockOccurrence(Methods.GET) }).toThrow(WSManErrors.UNSUPPORTED_METHOD)
+      expect(() => { castedIPSClass.AlarmClockOccurrence(Methods.PUT) }).toThrow(WSManErrors.UNSUPPORTED_METHOD)
+    })
+  })
   describe('ips_OptInService Tests', () => {
```
Comment (rsdmike) [inline src/ips/messages.test.ts:59]: Thank you for updating these to use a variable!

File: src/amt/models.ts [hunk 4]
```diff
@@ -352,3 +353,11 @@ export interface RemoteAccessPolicyAppliesToMPS extends CIM.Models.PolicySetAppl
   OrderOfAccess: number
   MpsType: number
 }
+
+export interface AlarmClockService extends CIM.Models.Service {
+  Name: string
```
Comment (rsdmike) [inline src/amt/models.ts:358]: I think you can remove all the properties here, they exist in the types they extend

File: src/amt/models.ts [hunk 5]
```diff
@@ -5,6 +5,7 @@
 
 import * as CIM from '../cim'
 import { NetworkPortConfigurationService } from '../cim/models'
+import { AlarmClockOccurrence } from '../ips/models'
```
Comment (rsdmike) [inline src/amt/models.ts:8]: My linter is showing this isn't used.

**Review by rsdmike — COMMENTED** [review]
Looking good, just a few changes. If you could resolve the conflict in package-lock and squash your commits, I think we can merge this in. 

EDIT: i was able to resolve the package-json for ya

File: src/amt/models.ts (hunk 4 above)
Comment (orinem) [reply src/amt/models.ts:358]: I'll fix the duplicate variables today.  Thanks for the review.

**Comment by orinem** [comment]
@rsdmike - I fixed the duplicate properties/unneeded import and squashed it down to one commit.

**Review by rsdmike — APPROVED** [review]

**Pull request merged**